cassovary-core/src/main/scala/com/twitter/cassovary/graph/Traverser.scala

```diff
@@ -24,21 +24,21 @@ import scala.util.Random
 /**
  * A Traverser traverses the graph in a certain order of nodes.
  */
-trait Traverser extends Iterator[Node] { self =>
+trait Traverser[+V <: Node] extends Iterator[V] { self =>
   /**
    * Traverser only visits node Ids listed in some node's edges, and thus
    * we know that node must exist, safe to assume graph.getNodeById return non-None value
    */
-  protected def getExistingNodeById(graph: Graph, id: Int) = graph.getNodeById(id).get
+  protected def getExistingNodeById[V <: Node](graph: Graph[V], id: Int) = graph.getNodeById(id).get
```

Comment:
By defining a new [V <: Node] here you have shadowed the V in Traverser[V]

what you actually want here is:
getExistingNodeById[W >: V](graph: Graph[W]...)


see http://docs.scala-lang.org/tutorials/tour/variances.html